packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, createElement, Dispatch, ReactElement, useMemo, useReducer, useRef } from "react";
+
+import AccordionGroup, { AccGroup } from "./AccordionGroup";
+import { AccordionGroupsReducerAction, getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
+import { AccordionContainerProps } from "../../typings/AccordionProps";
+
+export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
+    id: string;
+    groups: AccGroup[];
+    collapsible: boolean;
+    singleExpandedGroup?: boolean;
+}
+
+export const AccordionGroupsDispatch = createContext<Dispatch<AccordionGroupsReducerAction> | undefined>(undefined);
+
+export default function Accordion(props: AccordionProps): ReactElement | null {
+    const previousGroupsPropValue = useRef(props.groups);
+
+    const [accordionGroups, accordionGroupsDispatch] = useReducer(
+        getAccordionGroupsReducer(props.collapsible ? (props.singleExpandedGroup ? "single" : "multiple") : "all"), // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
+        props.groups.map(group => (props.collapsible ? { ...group, collapsed: true } : { ...group, collapsed: false }))
+    );
+
+    if (props.groups !== previousGroupsPropValue.current) {
+        previousGroupsPropValue.current = props.groups;
+        accordionGroupsDispatch({ type: "sync", groups: props.groups });
+    }
+
+    const renderedGroups = useMemo(() => {
+        return accordionGroups.map((group, index) => <AccordionGroup key={index} group={group} />);
+    }, [accordionGroups]);
```

Comment:
This feels a bit overkill, did you come across issues with this? Because I would assume that the `key` would be enough to prevent this from re-rendering.

Reply:
Removed the useMemo.